tools/cli/commands/rsync.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * External dependencies
+ */
+import inquirer from 'inquirer';
+import chalk from 'chalk';
+import watch from 'node-watch';
+import process from 'process';
+
+/**
+ * Internal dependencies
+ */
+import { allProjectsByType, dirs } from '../helpers/projectHelpers.js';
+import { runCommand } from '../helpers/runCommand.js';
+import { projectDir } from '../helpers/install.js';
+
+/**
+ * Entry point for the CLI.
+ *
+ * @param {object} argv - The argv for the command line.
+ */
+export async function rsyncInit( argv ) {
+	argv = await promptForPlugin( argv );
+	argv = await promptForDest( argv );
+	const sourcePluginPath = projectDir( `plugins/${ argv.plugin }` );
+
+	if ( true === argv.watch ) {
+		console.log( chalk.yellow( `Watching for changes in ${ argv.plugin } to rsync` ) );
+		watch( sourcePluginPath, { recursive: true }, ( eventType, fileName ) => {
+			if ( fileName && 'update' === eventType ) {
+				// console.log( chalk.yellow( `Changed detected in ${ fileName }...` ) );
+				rsyncToDest( sourcePluginPath, argv.dest );
+			}
+		} );
+	} else {
+		await rsyncToDest( sourcePluginPath, argv.dest );
+	}
+}
+
+/**
+ * Function that does the actual work of rsync.
+ *
+ * @param {string} source - Source path.
+ * @param {string} dest - Destination path.
+ * @returns {Promise<void>}
+ */
+async function rsyncToDest( source, dest ) {
+	try {
+		await runCommand( 'rsync', [
+			`-azLKP --delete --delete-after`,
+			"--exclude='jetpack_vendor/**/vendor' --exclude='node_modules' --exclude='vendor/**/vendor' --exclude='.cache' --exclude='tests' --exclude='*.map'",
+			source,
+			dest,
+		] );
```

Comment:
You forgot to change this to use the list of files from `listProjectFiles`.

Looks like `--files-from=-` that I suggested earlier may not quite work. It'll copy the files ok, but it seems it doesn't play well with `--delete` in that nothing will be deleted.

A command like `rsync -azLKP --prune-empty-dirs --delete --delete-after --delete-excluded --include=/**/ --include-from=- --exclude=* src/ dest/` (adapted from https://stackoverflow.com/a/1813972/634419) seems to work, although the list from `listProjectFiles` will need munging like `.map( v => '/' + ( v.match( /[[*?]/ ) ? v.replace( //[[*?\\]/g, '\\$&' ) : v ) )` before being written to the command's stdin.

Reply:
@anomiex thanks for the suggestions. No matter how I approached this, the `exclude=*` seemed to take precedent. I even ended up adding some tmp file creation and reading to satisfy the `include-from` suggestion. This last commit is the closest I was able to get to something satisfactory in terms of usability. If you have any other suggestions, I'm all ears.